### PR TITLE
Repeated fields support UI

### PR DESCRIPTION
**Pull request by ruairif**
Depends on: https://github.com/scrapinghub/portia/pull/563

### CODE REVIEW

File: portiaui/app/services/annotation-structure.js [hunk 1]
```diff
@@ -146,14 +147,39 @@ const ElementStructure = Ember.Object.extend({
...
+                                        }
+                                        if (i > 1) {
+                                            break;
+                                        }
+                                    }
+                                    if (i > 1) {
+                                        annotation.set('repeated', true);
+                                        break;
+                                    }
+                                }
```
Comment (sagelliv) [inline portiaui/app/services/annotation-structure.js]: This is more inefficient but is a bit more readable.

```
const isRepeated = repeatedContainers.map((container) => {
    return elements.filter((child) => { container.contains(child); });
}).any((repeated) => { repeated.length > 1 });

if (isRepeated) {
    annotation.set('repeated', true);
}
```

Comment (michalmo) [inline portiaui/app/services/annotation-structure.js]: that will give a jshint warning about functions in function :(

File: portiaui/app/services/annotation-structure.js [hunk 2]
```diff
@@ -146,14 +147,39 @@ const ElementStructure = Ember.Object.extend({
...
                             xpath: annotation.get('selectorGenerator.xpath')
                         });
                     }
+                    annotation.set('repeated', false);
                     if (selector) {
                         selectorMatcher.register(selector, setElements);
+                        const elements = selectorMatcher.query(selector);
+                        if (repeatedSelector && elements && elements.length > 1) {
+                            const repeatedContainers = annotation.get('containerGenerator.parent.repeatedContainers');
+                            if (repeatedContainers.length > 1) {
```
Comment (michalmo) [inline portiaui/app/services/annotation-structure.js]: this logic belongs in the selector generator. and we should just read repeated from it here

File: portiaui/app/services/annotation-structure.js (hunk 1 above)
Comment (sagelliv) [inline portiaui/app/services/annotation-structure.js]: Extract the inner function out?

`repeatedContainers.map(containedChildren).any((repeated) => { repeated.length > 1 });`

**Comment by sagelliv** [comment]
Rebase @ruairif

File: portiaui/app/components/annotation-options.js [hunk 3]
```diff
@@ -96,9 +97,42 @@ export default Ember.Component.extend({
...
+                return (this._updateSelectorPromise = this._updateSelector(sample));
+            });
+        }
+
+        return this._updateSelectorPromise;
+    },
+
+    _updateSelector(sample) {
+        const dispatcher = this.get('dispatcher');
+        const promise = dispatcher.updateSampleSelectors(sample).then(() => {
```
Comment (sagelliv) [inline portiaui/app/components/annotation-options.js]: Why not return here directly?

Comment (ruairif) [inline portiaui/app/components/annotation-options.js]: The promise needs to be in the closure that wraps the resolving function

**Pull request merged**